### ISSUE TITLE
Could i3-config-wizard be made scriptable?

### ISSUE THREAD

**Issue by yawpitch** [i3/i3#3136]
TL;DR: would it be possible to add a `--meta=win` or `--meta=alt` option flag to **i3-config-wizard** and write a config without any user graphical interaction being required?

The use case of this enhancement is fairly specific: I'm using a provisioning tool (in my case Ansible) to bootstrap user workstations, and I want to deploy -- without user interaction --- a basic I3 config file, which will then be further modified by the provisioning tool to create the site-wide base configuration for I3.

I know the user's keymap choice and meta key preference before the Ansible playbook runs, but the actual provisioning is done headless, so I cannot and do not want to hand control over to the i3-config-wizard part way through the run. I do, however, want the benefits of a keymap-aware I3 config created as if the user had run i3-config-wizard and manually selected their preferred meta key. Right now my only choice seems to be build a template for every possible keymap and have Ansible select and copy over the right one, but this is really quite a painful choice.

Since the only actual interaction the i3-config-wizard provides is a boolean choice of whether or not to create the config, and if that is True a choice of meta key, it doesn't seem that it would be too difficult to allow something like:

`i3-config-wizard --meta win`

And have it assume that the user does indeed want to write a config appropriate for their meta key choice, and write that without going through the trouble of ever opening a window or grabbing the keyboard.

I'd write up a pull request, but frankly my C skills are so rusty I'm sure I'd just embarass myself.

**Comment by Airblader** [comment]
I can see your usecase and I do think we should find a way to support this. However, I don't think modifying i3-config-wizard is the correct way to do so since it is explicitly designed to be a graphical wizard.

Perhaps the logic to do this could be moved into libi3 somehow?

**Comment by yawpitch** [comment]
From my perspective it doesn't really matter where it exists; the wizard seems a natural entry point for me, as it's already a command line invoked tool that contains all the logic for binding keymap keys to configuration file values. I do understand that it's current use is to provide a GUI, but I'd argue that it's explicit goal is to produce the config keymap-appropriate config file, and that providing a wizard for asking for a Meta key selection is secondary to that, and the flag I'm describing would just be an alternate way of providing an answer to that question.

That said I'd be completely happy with a distinct CLI tool for this purpose, with the core binding logic moved to libi3 and both the new tool and the wizard using that lib.

Thank you for considering it.

M

**Comment by Airblader** [comment]
I'll move this to accepted assuming a reasonable implementation. That said, I won't be working on this myself.

**Comment by yawpitch** [comment]
Fair enough. If no one picks it up soonish I'll see if I can dust off enough C to pull together a not ridiculously bad attempt at it.

**Comment by stapelberg** [comment]
I think the wizard is actually an okay place for this to live. The behavior is not general enough to warrant being in libi3 (yet?), so let’s go with the suggested command-line flag.

**Comment by ograff** [comment]
I'd like to take this if no one has yet.

**Comment by Airblader** [comment]
Go ahead!
